modules/crypto/build.gradle

```diff
@@ -15,12 +15,17 @@ thirdPartyAudit.enabled = false
 forbiddenApisTest.ignoreFailures = true
 testingConventions.enabled = false
 
+opensearchplugin {
+  description 'Crypto module plugin for providing encryption and decryption support.'
+  classname 'org.opensearch.encryption.CryptoModulePlugin'
+}
+
 dependencies {
   // Common crypto classes
   api project(':libs:opensearch-common')
 
-  // Encryption
-  implementation "com.amazonaws:aws-encryption-sdk-java:2.4.0"
+  implementation "com.amazonaws:aws-encryption-sdk-java:1.7.0"
```

Comment:
Unrelated, but why we are using version that was release in 2020? https://mvnrepository.com/artifact/com.amazonaws/aws-encryption-sdk-java/1.7.0

Reply:
https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/migration-guide.html Check this

Reply:
Please check this:

![image](https://github.com/opensearch-project/OpenSearch/assets/509855/8658070e-9ea0-40cb-98c7-b1bfa6806322)

There are CVEs reported

Reply:
First 2 are coming from bouncy castle and third one is coming from junit. I haven't used that dependency. I am using bcprov-jdk15to18 v1.75 which doesn't have a vulnerability.

Reply:
First 2 are coming from bouncy castle (bcprov-ext-jdk15on) and third one is coming from junit. I haven't used that dependency. I am using bcprov-jdk15to18 v1.75 which doesn't have a vulnerability.
The direct vulnerability is actually a feature addition to strengthen the security. In order to adopt this, SDK needs to be migrated to 2.x. This can only be done if data encrypted by previous versions <1.7 have been ported to 1.7 first.

Reply:
We don't have data encrypted with previous version <1.7, do we?

Reply:
In OS we don't. If someone has customization done in BlobContainer implementations to encrypt data then they might be encrypting from previous versions.

Reply:
> If someone has customization done in BlobContainer implementations to encrypt data then they might be encrypting from previous versions.

This is not the OpenSearch concern,  if someone has customization to encrypt, she definitely has the customization to decrypt. We should use the latest 2.4.1 I think.

Reply:
This is not about customization in decrypt because of customization in encrypt. This is about handling encrypted data of users who didn't have a choice then because OS didn't have encryption support and had to build their own encryption layer. OpenSearch bringing encryption now shouldn't force this on such users and just leave them with choice of either reindexing their data or ignoring the encryption altogether which might mean incompatible dependency issues in which case this option will no longer be valid. 

 It should be OpenSearch's concern to gracefully allow these users to migrate to newer versions especially when it can do so. AWS ESDK has v1.7 rolled out for this specific purpose and is both forward and backward compatible. OS can decide to upgrade SDK to 2.x in a major version upgrade (say 4.0) which would the right step aligned with major version bump property of introducing a breaking change.

Reply:
@vikasvb90 we are talking about one specific library, it is not feasible to know, more to support, what kind of mechanisms or algorithms users could have been used for encryption / decryption of their data prior to whatever we would be introducing in OpenSearch. To your point,. we surely could help them and I have no issues supporting 1.x release line **if it is maintained**, so far it does not look like. People flood us with CVEs same day they are released, if this library + version is going to be flagged by security scanning tools (which it seems like it will) - this is no go.